mlflow/models/evaluation/base.py

```diff
@@ -365,7 +366,7 @@ def __init__(self, data, *, targets, name=None, path=None, feature_names=None):
                 self._feature_names = feature_names
             else:
                 self._features_data = data.drop(targets, axis=1, inplace=False)
-                self._feature_names = list(self._features_data.columns)
+                self._feature_names = [stringify_column(c) for c in self._features_data.columns]
```

Comment:
```suggestion
                self._feature_names = [str(c) for c in self._features_data.columns]
```

Why not just apply `str`?

Reply:
I was just following this suggestion: https://github.com/mlflow/mlflow/pull/6555#issuecomment-1230084981

Reply:
Got it. @WeichenXu123 Why should we use `feature_{str(raw_column_name)}`?

Reply:
To clarify, I didn't mean rename column name.

My previous comment say, if the dataframe column has the column name, do not change it, but if the dataframe column is unlabeled, then auto generate the column name by `feature_{index}` to make it clear. And the column name is only used in shap plot display, we should not use the column name while running prediction.

Reply:
Makes sense! Thanks for the explanation.

Reply:
I think `stringify_column` is confusing. It sounds like it just applies `str`, but it doesn't. Can we use a name such as `generate_feature_name_if_not_string`?